### PR TITLE
Fix a few typos

### PR DESCRIPTION
**Pull request by Nielius**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,14 +25,14 @@ The symbolic automaton generalizes the classic automaton's transition to a predi
 state_state -> next_state if pred(input), where pred = \input -> input == 'a'
 ```
 
-This has implications that we could explore the automaton without requiring to know the full alphabet and in fact the alphabet can now be infinite, which sacrificing any decidability properties. Unfortunately this does push some complexity onto the transitions, which can grow into `if` expressions with exponential complexity.
+This has the implication that we could explore the automaton without knowing the full alphabet and in fact the alphabet can now be infinite, without sacrificing any decidability properties. Unfortunately this does push some complexity onto the transitions, which can grow into `if` expressions with exponential complexity.
```
Comment (Nielius) [inline README.md:28]: Is the `which` -> `without` change correct?

File: README.md [hunk 2]
```diff
@@ -25,14 +25,14 @@ The symbolic automaton generalizes the classic automaton's transition to a predi
...
 
-In the final implementation of the algorithm, these predicates can be a set of nested function, for example: `and(ge($int, 10), lt($int, 20))`, which we can also explore, where we could prove that evaluation during compilation of the constant parts are equivalent to evaluation at runtime.
+In the final implementation of the algorithm, these predicates can be a set of nested functions, for example: `and(ge($int, 10), lt($int, 20))`. This is also something we can explore, and we could prove that evaluation during compilation of the constant parts is equivalent to evaluation at runtime.
 
-The reason to explore symbolic automaton first, is because symbolic derivatives can technically use trees as its input alphabet and so provide a quick win, but also we will also heavily use the `if` expressions in our final VPA implementation.
+The reason to explore symbolic automaton first, is because symbolic automata can technically use trees as their input alphabet and so provide a quick win, but also we will also heavily use the `if` expressions in our final VPA implementation.
```
Comment (Nielius) [inline README.md:35]: Is the `derivatives` -> `automata` change correct?

**Comment by Nielius** [comment]
Mostly these are just small typos that I happened to see when I was reading, but there are two cases (I've commented which ones) where the meaning changes a bit.

File: README.md (hunk 2 above)
Comment (awalterschulze) [reply README.md:35]: their is good

**Comment by awalterschulze** [comment]
Except for derivatives -> automata everything else looks good in this pull request.  Thank you so much for all the corrections

**Comment by Nielius** [comment]
Ok, fixed that.

**Pull request merged**